### ISSUE TITLE
Can't mix LinVST and yabridge plugins (WINEPREFIX problem)

### ISSUE THREAD

**Issue by lentferj** [robbert-vdh/yabridge#28]
I just tried to load an old session in Mixbus 32C v6.1. This session has a few plugins still emulated via  LinVST (Waves) and the rest is emulated via yabridge.

I run into problems when loading yabridge emulated plugins AFTER loading the Waves plugin. It seems that LinVST changes/sets the WINEPREFIX permanently. This causes problems at least with plugins searching for extra files in the Prefix, as in this case for Amplitube 4.

I know this is likely a misbehaviour of LinVST, but as long as Waves do not work, I cannot switch entirely.
Any way to work around this? 

15:16:18 [TB_Barricade_v4-3AWceknx] Initializing yabridge version 1.4.1
15:16:18 [TB_Barricade_v4-3AWceknx] host:         '/home/lentferj/.local/share/yabridge/yabridge-host.exe'
15:16:18 [TB_Barricade_v4-3AWceknx] plugin:       '/home/lentferj/.wine64/drive_c/Program Files/VstPlugins/Toneboosters/Barricade_v4/TB_Barricade_v4.dll'
15:16:18 [TB_Barricade_v4-3AWceknx] realtime:     'yes'
15:16:18 [TB_Barricade_v4-3AWceknx] socket:       '/tmp/yabridge-TB_Barricade_v4-3AWceknx.sock'
**_15:16:18 [TB_Barricade_v4-3AWceknx] wine prefix:  '/home/lentferj/.wine64'_**
15:16:18 [TB_Barricade_v4-3AWceknx] wine version: '5.8 (Staging)'
15:16:18 [TB_Barricade_v4-3AWceknx] 
15:16:18 [TB_Barricade_v4-3AWceknx] config from:   '<defaults>'
15:16:18 [TB_Barricade_v4-3AWceknx] hosting mode:  'individually, 64-bit'
15:16:18 [TB_Barricade_v4-3AWceknx] other options: '<none>'
15:16:18 [TB_Barricade_v4-3AWceknx] 
15:16:18 [TB_Barricade_v4-3AWceknx] Enabled features:
15:16:18 [TB_Barricade_v4-3AWceknx] - bitbridge support
15:16:18 [TB_Barricade_v4-3AWceknx] 
15:16:18 [TB_Barricade_v4-3AWceknx] [Wine STDOUT] Initializing yabridge host version 1.4.1
15:16:19 [TB_Barricade_v4-3AWceknx] [Wine STDOUT] Finished initializing '/home/lentferj/.wine64/drive_c/Program Files/VstPlugins/Toneboosters/Barricade_v4/TB_Barricade_v4.dll'
Emulate VST Bypass Port for TB_Barricade_v4
\[...\]
Loading  /home/lentferj/.wine64_waves/drive_c/Program Files/VSTPlugIns/WaveShell1-VST 10.0_x64.dll
00bc:fixme:win:RegisterTouchWindow (0x2005a 00000002): stub
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
00bc:fixme:nls:GetThreadPreferredUILanguages 00000038, 0x21bf84, 0x21bfa0 0x21bf80
00bc:fixme:nls:get_dummy_preferred_ui_language (0x38 0x21bf84 0x21bfa0 0x21bf80) returning a dummy value (current locale)
00bc:fixme:heap:RtlSetHeapInformation 0x3a50000 0 0x21a940 4 stub
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
00cc:fixme:ver:GetCurrentPackageId (00000000043CFDA0 0000000000000000): stub
00cc:fixme:sync:SetWaitableTimerEx (00000000000000AC, 00000000043CFC50, 0, 0000000000000000, 0000000000000000, 0000000000000000, 32) semi-stub
Do *not* Emulate VST Bypass Port for Scheps 73 Stereo
Emulate VST Bypass Port for AVA LegacyQ
effSetBypass 1
setNumInputs ( 4 );
setNumOutputs ( 2 );
allocateArrangement ( &plugInput, 2 );
allocateArrangement ( &plugOutput, 2 );
Emulate VST Bypass Port for EQ500
effSetBypass 1
DSSI VST plugin server v0.986
Copyright (c) 2012-2013 Filipe Coelho
Copyright (c) 2010-2011 Kristian Amlie
Copyright (c) 2004-2006 Chris Cannam
LinVst version 2.8.0
Loading  /home/lentferj/.wine64_waves/drive_c/Program Files/VSTPlugIns/WaveShell1-VST 10.0_x64.dll
00dc:fixme:win:RegisterTouchWindow (0x2007c 00000002): stub
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
00dc:fixme:nls:GetThreadPreferredUILanguages 00000038, 0x21bf84, 0x21bfa0 0x21bf80
00dc:fixme:nls:get_dummy_preferred_ui_language (0x38 0x21bf84 0x21bfa0 0x21bf80) returning a dummy value (current locale)
00dc:fixme:heap:RtlSetHeapInformation 0x3a50000 0 0x21a940 4 stub
wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
00ec:fixme:ver:GetCurrentPackageId (00000000043CFDA0 0000000000000000): stub
00ec:fixme:sync:SetWaitableTimerEx (00000000000000B0, 00000000043CFC50, 0, 0000000000000000, 0000000000000000, 0000000000000000, 32) semi-stub
Do *not* Emulate VST Bypass Port for Scheps 73 Stereo
15:16:47 [Head Crusher-7CDJKRTa] Initializing yabridge version 1.4.1
15:16:47 [Head Crusher-7CDJKRTa] host:         '/home/lentferj/.local/share/yabridge/yabridge-host.exe'
15:16:47 [Head Crusher-7CDJKRTa] plugin:       '/home/lentferj/.wine64/drive_c/Program Files/Steinberg/VstPlugins/Head Crusher.dll'
15:16:47 [Head Crusher-7CDJKRTa] realtime:     'yes'
15:16:47 [Head Crusher-7CDJKRTa] socket:       '/tmp/yabridge-Head Crusher-7CDJKRTa.sock'
**_15:16:47 [Head Crusher-7CDJKRTa] wine prefix:  '/home/lentferj/.wine64_waves \<overridden\>'_**
15:16:47 [Head Crusher-7CDJKRTa] wine version: '5.8 (Staging)'
15:16:47 [Head Crusher-7CDJKRTa] 
15:16:47 [Head Crusher-7CDJKRTa] config from:   '<defaults>'
15:16:47 [Head Crusher-7CDJKRTa] hosting mode:  'individually, 64-bit'
15:16:47 [Head Crusher-7CDJKRTa] other options: '<none>'
15:16:47 [Head Crusher-7CDJKRTa] 
15:16:47 [Head Crusher-7CDJKRTa] Enabled features:
15:16:47 [Head Crusher-7CDJKRTa] - bitbridge support
15:16:47 [Head Crusher-7CDJKRTa] 
15:16:48 [Head Crusher-7CDJKRTa] [Wine STDOUT] Initializing yabridge host version 1.4.1
15:16:49 [Head Crusher-7CDJKRTa] [Wine STDOUT] Finished initializing '/home/lentferj/.wine64/drive_c/Program Files/Steinberg/VstPlugins/Head Crusher.dll'
\[...\]
15:17:03 [AmpliTube 4-5HJKZhmt] Initializing yabridge version 1.4.1
15:17:03 [AmpliTube 4-5HJKZhmt] host:         '/home/lentferj/.local/share/yabridge/yabridge-host.exe'
15:17:03 [AmpliTube 4-5HJKZhmt] plugin:       '/home/lentferj/.wine64_TrackS/drive_c/Program Files/VstPlugIns/AmpliTube 4.dll'
15:17:03 [AmpliTube 4-5HJKZhmt] realtime:     'yes'
15:17:03 [AmpliTube 4-5HJKZhmt] socket:       '/tmp/yabridge-AmpliTube 4-5HJKZhmt.sock'
**_15:17:03 [AmpliTube 4-5HJKZhmt] wine prefix:  '/home/lentferj/.wine64_waves \<overridden\>'_**
15:17:03 [AmpliTube 4-5HJKZhmt] wine version: '5.8 (Staging)'
15:17:03 [AmpliTube 4-5HJKZhmt] 
15:17:03 [AmpliTube 4-5HJKZhmt] config from:   '<defaults>'
15:17:03 [AmpliTube 4-5HJKZhmt] hosting mode:  'individually, 64-bit'
15:17:03 [AmpliTube 4-5HJKZhmt] other options: '<none>'
15:17:03 [AmpliTube 4-5HJKZhmt] 
15:17:03 [AmpliTube 4-5HJKZhmt] Enabled features:
15:17:03 [AmpliTube 4-5HJKZhmt] - bitbridge support
15:17:03 [AmpliTube 4-5HJKZhmt] 
15:17:03 [AmpliTube 4-5HJKZhmt] [Wine STDOUT] Initializing yabridge host version 1.4.1
15:17:03 [AmpliTube 4-5HJKZhmt] [Wine STDERR] 0118:fixme:heap:RtlSetHeapInformation 0x5a0000 0 0x21eca0 4 stub

**Comment by robbert-vdh** [comment]
Looks like LinVST sets the `WINEPREFIX` environment variable for the Mixbus process rather than for the host application it's trying to launch, which could of course break all kinds of things. What's the issue with Waves plugins? Fixing whatever doesn't work there would probably be the best solution here. I've tried looking at that before but I couldn't get their software installed because it requires a specific version of powershell. How did you manage to install it? I'm away from my desktop for the week, but I'll gladly take a look when I get back.

**Comment by lentferj** [comment]
I did install Waves as described on the LinVST README. No problems here and I can't remember I needed to do anything about Powershell.

**Comment by robbert-vdh** [comment]
The Waves Central installer errors out with:
![Screenshot_20200811_210513](https://user-images.githubusercontent.com/748520/89938129-5c0d7b00-dc16-11ea-8e6b-d0e2e499d27c.png)
I've tried installing an older version of PowerShell with the winetricks verb from [this thread](https://forum.winehq.org/viewtopic.php?f=8&t=33247), which seems to be the only up to date thread on the subject. That got me through the installer, but now Waves Central itself throws the same error. I'd love to see what's wrong with the actual plugins, but it seems like the installer just doesn't work at the moment.

**Comment by lentferj** [comment]
This is the console output  I see when I try to load a Wave plugin with yabridge in MB 32c v61

21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] Initializing yabridge version 1.4.1
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] host:         '/home/lentferj/.local/share/yabridge/yabridge-host.exe'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] plugin:       '/home/lentferj/.wine64_waves/drive_c/Program Files/VSTPlugIns/WaveShell1-VST 10.0_x64.dll'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] realtime:     'yes'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] socket:       '/tmp/yabridge-WaveShell1-VST 10.0_x64-27FWYact.sock'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] wine prefix:  '/home/lentferj/.wine64_waves'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] wine version: '5.8 (Staging)'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] 
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] config from:   '<defaults>'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] hosting mode:  'individually, 64-bit'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] other options: '<none>'
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] 
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] Enabled features:
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] - bitbridge support
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] 
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 002c:fixme:winediag:__wine_start_process Wine Staging 5.8 is a testing version containing experimental patches.
21:41:25 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 002c:fixme:winediag:__wine_start_process Please mention your exact version when filing bug reports on winehq.org.
21:41:26 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 0078:err:module:load_so_dll failed to load .so lib "/opt/wine-staging/lib64/wine/wineusb.sys.so": /opt/wine-staging/lib64/wine/wineusb.sys.so: undefined symbol: libusb_interrupt_event_handler
21:41:26 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 0078:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\wineusb": c0000142
21:41:26 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 003c:fixme:service:scmdatabase_autostart_services Auto-start service L"wineusb" failed to start: 1114
21:41:26 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 002c:fixme:ver:GetCurrentPackageId (000000000031FA80 0000000000000000): stub
21:41:26 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDOUT] Initializing yabridge host version 1.4.1
21:41:27 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 00bc:fixme:win:RegisterTouchWindow (0x20050 00000002): stub
21:41:27 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
21:41:27 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
21:41:27 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 00bc:fixme:nls:GetThreadPreferredUILanguages 00000038, 0x21cd84, 0x21cda0 0x21cd80
21:41:27 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 00bc:fixme:nls:get_dummy_preferred_ui_language (0x38 0x21cd84 0x21cda0 0x21cd80) returning a dummy value (current locale)
21:41:28 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
21:41:28 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDOUT] Finished initializing '/home/lentferj/.wine64_waves/drive_c/Program Files/VSTPlugIns/WaveShell1-VST 10.0_x64.dll'
21:41:28 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 00bc:fixme:heap:RtlSetHeapInformation 0x3d90000 0 0x21ce20 4 stub
21:41:28 [WaveShell1-VST 10.0_x64-27FWYact] [Wine STDERR] 00bc:fixme:win:RegisterTouchWindow (0x20082 00000002): stub

**Comment by lentferj** [comment]
I ttried some more and some actually seem to work :D

![grafik](https://user-images.githubusercontent.com/5664183/90180382-294aba80-ddaf-11ea-9610-0fc0cbf8d1e0.png)


here is the console ouput from this one
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] Initializing yabridge version 1.4.1
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] host:         '/home/lentferj/.local/share/yabridge/yabridge-host.exe'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] plugin:       '/home/lentferj/.wine64_waves/drive_c/Program Files/VSTPlugIns/WaveShell1-VST 10.0_x64.dll'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] realtime:     'yes'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] socket:       '/tmp/yabridge-WaveShell1-VST 10.0_x64-02EHTadg.sock'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] wine prefix:  '/home/lentferj/.wine64_waves'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] wine version: '5.8 (Staging)'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] 
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] config from:   '<defaults>'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] hosting mode:  'individually, 64-bit'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] other options: '<none>'
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] 
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] Enabled features:
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] - bitbridge support
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] 
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 002c:fixme:winediag:__wine_start_process Wine Staging 5.8 is a testing version containing experimental patches.
21:50:40 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 002c:fixme:winediag:__wine_start_process Please mention your exact version when filing bug reports on winehq.org.
21:50:41 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 0078:err:module:load_so_dll failed to load .so lib "/opt/wine-staging/lib64/wine/wineusb.sys.so": /opt/wine-staging/lib64/wine/wineusb.sys.so: undefined symbol: libusb_interrupt_event_handler
21:50:41 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 0078:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\wineusb": c0000142
21:50:41 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 003c:fixme:service:scmdatabase_autostart_services Auto-start service L"wineusb" failed to start: 1114
21:50:41 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 002c:fixme:ver:GetCurrentPackageId (000000000031FA80 0000000000000000): stub
21:50:41 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDOUT] Initializing yabridge host version 1.4.1
21:50:42 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00bc:fixme:win:RegisterTouchWindow (0x20050 00000002): stub
21:50:42 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
21:50:42 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
21:50:42 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00bc:fixme:nls:GetThreadPreferredUILanguages 00000038, 0x21cd84, 0x21cda0 0x21cd80
21:50:42 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00bc:fixme:nls:get_dummy_preferred_ui_language (0x38 0x21cd84 0x21cda0 0x21cd80) returning a dummy value (current locale)
21:50:42 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00bc:fixme:heap:RtlSetHeapInformation 0x4530000 0 0x21b740 4 stub
21:50:42 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00bc:fixme:heap:RtlSetHeapInformation 0x4640000 0 0x21b740 4 stub
21:50:43 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] wine: Read access denied for device L"\\??\\Z:\\", FS volume label and serial are not available.
21:50:43 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDOUT] Finished initializing '/home/lentferj/.wine64_waves/drive_c/Program Files/VSTPlugIns/WaveShell1-VST 10.0_x64.dll'
21:50:43 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00cc:fixme:ver:GetCurrentPackageId (000000000679FDA0 0000000000000000): stub
21:50:43 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00cc:fixme:sync:SetWaitableTimerEx (00000000000000AC, 000000000679FC50, 0, 0000000000000000, 0000000000000000, 0000000000000000, 32) semi-stub
21:50:45 [WaveShell1-VST 10.0_x64-02EHTadg] [Wine STDERR] 00bc:fixme:win:RegisterTouchWindow (0x20082 00000002): stub

**Comment by robbert-vdh** [comment]
For that first plugin, could you start Mixbus with the command below and then try loading the plugin again? That will give some more information on what the plugin is actually doing that might cause Mixbus to freeze. The output will be written to `/tmp/yabridge.log` and there'll probably be a lot of it.

```shell
env YABRIDGE_DEBUG_FILE=/tmp/yabridge.log YABRIDGE_DEBUG_LEVEL=2 <however_you_start_mixbus_on_your_system>
```

Also, have you tried to load the plugin in another VST host (e.g. Bitwig or Carla)? Do the plugins behave similarly there?

**Comment by lentferj** [comment]
Here is the log file from trying to load API2500 Mono plugin onto a mono audio track
[yabridge.log](https://github.com/robbert-vdh/yabridge/files/5071208/yabridge.log)

**Comment by lentferj** [comment]
on Bitwig the same plugin loaded and opened.... but when I tried closing the gui and froze and crashed. Logfile attached.
[yabridge.log](https://github.com/robbert-vdh/yabridge/files/5071281/yabridge.log)

**Comment by lentferj** [comment]
ha - so, after waiting 4 minutes, the API2500 actually loaded in MB32C, too... here are the last lines of yabridge.log. Maybe you can see what took it so long :)

22:31:30 [WaveShell1-VST 10.0_x64-58CFPYav]    audioMasterCallback() :: 1
22:31:30 [WaveShell1-VST 10.0_x64-58CFPYav] >> dispatch() effEditGetRect(index = 0, value = 0, option = 0, data = <writable_buffer>)


22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav]    dispatch() :: 1, {l: 0, t: 0, r: 690, b: 393}
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav] >> dispatch() effEditIdle(index = 0, value = 0, option = 0, data = <nullptr>)
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav]    dispatch() :: 0
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav] >> dispatch() effGetChunk(index = 0, value = 0, option = 0, data = <writable_buffer>)
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav] >> dispatch() effEditIdle(index = 0, value = 0, option = 0, data = <nullptr>)
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav]    dispatch() :: 794, <794 byte chunk>
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav] >> dispatch() effGetProgram(index = 0, value = 0, option = 0, data = <nullptr>)
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav]    dispatch() :: 0
22:35:34 [WaveShell1-VST 10.0_x64-58CFPYav]    dispatch() :: 0
22:35:35 [WaveShell1-VST 10.0_x64-58CFPYav] >> dispatch() effEditIdle(index = 0, value = 0, option = 0, data = <nullptr>)

**Comment by lentferj** [comment]
That was a short period of happines.. after 2 min it froze again. Full log attached.

[yabridge.log](https://github.com/robbert-vdh/yabridge/files/5071318/yabridge.log)

**Comment by robbert-vdh** [comment]
Interesting. So Mixbus freezes when it asks the plugin for the dimensions of the editor after the editor has been opened and is stuck waiting indefinitely for a response, and Bitwig could not even ask the plugin to close its editor because it got stuck at `effEditIdle()` (which shouldn't really do anything other than tell the plugin that it should refresh the GUI in the next message loop). My guess is that the plugin is doing some weird stuff and is stuck in the Win32 message loop (that's how Windows applications handle events, and because it's a pretty ancient system it all has to be single threaded so no other events can be processed while the message loop is running).

If you'd care to test my hypothesis, try running the Waves plugin in a yabridge plugin group. When using plugin groups the message loop only handles 20 events at once to prevent badly coded or heavy plugins from starving out the rest of the group). If that works, then we should just add a similar cap to the regular plugins (I've never across a plugin where this was needed before this). To set this up, create `~/.wine64_waves/drive_c/Program Files/VSTPlugIns/yabridge.toml` with the following contents (assuming that you're using `~/.wine64_waves/drive_c/Program Files/VSTPlugIns/WaveShell1-VST 10.0_x64.so` of course):

```toml
["WaveShell1-VST 10.0_x64.so"]
group = "waves"
```

And then load the plugin again.

**Comment by lentferj** [comment]
That actually did the trick. Thanks so much.
Initial loading in Bitwig still took a bit of time but once loaded it worked fine.
Mixbus loads fast and works.

Is it advisable to put each "WaveShellX-VST...so" on its own group, e.g. waves1, waves2, etc or should I just put all the WavesShellX in one group "waves"?

**Comment by robbert-vdh** [comment]
Great! So that would imply that Waves plugins are just doing some weird things that would cause them to get stuck in an infinite message loop. I've added the same limit to regular individually hosted plugins that was already being used in plugin groups, so those Waves plugins should now also work fine without any manual configuration. You can grab a build with this fix included from the [automated builds](https://github.com/robbert-vdh/yabridge/actions?query=workflow%3A%22Automated+builds%22+branch%3Amaster) page.

For those `WaveShell*.dll` files you could put them all in a single group by using `"WaveShell1-VST*so"`, but with this fix they should also run fine without using plugin groups.

**Comment by lentferj** [comment]
Looks very, very good...
![Bildschirmfoto vom 2020-08-14 14-07-03](https://user-images.githubusercontent.com/5664183/90247797-ac175800-de37-11ea-8c26-5da564ec4288.png)

**Comment by robbert-vdh** [comment]
Then it was just the plugins being a bit buggy! (or it was a compatibility issue with Wine, but having used Waves plugins in the past I'm going to bet on the former haha! Shame I couldn't test the plugins out for myself since Waves Central 11 just doesn't run under Wine. Let me know if you run into any other issues.

**Comment by lentferj** [comment]
Sure, will

**Comment by kwalker99** [comment]
Robbert, you could try Waves v10 for testing, the installer doesn’t use PowerShell...

Kind regards

Kevin Walker

On 14 Aug 2020, at 17:05, Jan Lentfer <notifications@github.com> wrote:

Closed #28.
